components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,352 @@
+		end,
+		row = 'localprize',
+		rowParse = function (prizePool, input, context, index)
+			return prizePool:_parseInteger(input)
+		end
+	},
+	[prizeTypes.QUALIFIES] = {
+		header = 'qualifies',
+		headerParse = function (prizePool, input, context, index)
+			local link = input:gsub(' ', '_')
+			local data = {link = link}
+
+			-- Automatically retrieve information from the Tournament
+			local tournamentData = prizePool:_getTournamentInfo(link)
+			if tournamentData then
+				data.title = tournamentData.tickername
+				data.icon = tournamentData.icon
+				data.iconDark = tournamentData.icondark
+			end
+
+			-- Manual override
+			local prefix = 'qualifies' .. index
+			data.title = context[prefix .. 'name'] or data.title
+
+			return data
+		end,
+		row = 'qualified',
+		rowParse = function (prizePool, input, context, index)
+			return Logic.readBool(input)
+		end
+	},
+	[prizeTypes.POINTS] = {
+		header = 'points',
+		headerParse = function (prizePool, input, context, index)
+			local pointsData = mw.loadData('Module:Points/data')
+			return pointsData[input] or {title = 'Points'}
+		end,
+		row = 'points',
+		rowParse = function (prizePool, input, context, index)
+			return prizePool:_parseInteger(input)
+		end
+	},
+	[prizeTypes.FREETEXT] = {
+		header = 'freetext',
+		headerParse = function (prizePool, input, context, index)
+			return {title = input}
+		end,
+		row = 'freetext',
+		rowParse = function (prizePool, input, context, index)
+			return input
+		end
+	},
+}
+
+local specialData = {
+	[specialDataTypes.GROUPSCORE] = {
+		field = 'wdl',
+		parse = function (prizePool, input, context)
+			return input
+		end
+	},
+	[specialDataTypes.LASTVS] = {
+		field = 'lastvs',
+		parse = function (prizePool, input, context)
+			return prizePool:_parseOpponentArgs(input, context.date)
+		end
+	},
+	[specialDataTypes.LASTVSSCORE] = {
+		field = 'lastvsscore',
+		parse = function (prizePool, input, context)
+			local scores = Table.mapValues(mw.text.split(input, '-'), tonumber)
+			return {score = scores[1], vsscore = scores[2]}
+		end
+	},
+}
+
+function PrizePool:init(args)
+	self.args = args
+	self.pagename = mw.title.getCurrentTitle().text
+	self.args.type = Json.parseIfString(self.args.type)
+	self.date = args.date or Variables.varDefaultMulti('tournament_enddate', 'tournament_edate', 'edate', TODAY)
+	if args.opponentLibrary then
+		Opponent = require('Module:'.. args.opponentLibrary)
+	end
+
+	if not self.args.type then
+		return error('Please provide a type!')
+	elseif type(self.args.type) ~= 'table' or not self.args.type.type then
+		return error('Could not parse type!')
+	elseif not Opponent.isType(self.args.type.type) then
+		return error('Not a valid type!')
+	end
+
+	self.opponentType = self.args.type.type
+
+	return self
+end
+
+function PrizePool:readInput(args)
+	self.options = self:_readStandardOptions(args)
+	self.prizes = self:_readPrizes(args)
+	self.placements = self:_readPlacements(args)
```

Comment:
ideally sc/sc2 would like to add (up to 2) additional options (inject) before we read placements
could be done in a later PR though (when we add opponent automation)

Reply:
Doesn't seem unreasonable. Which options?

Reply:
`hasArchonOpponent` (needed in display later on)
`importlimit` (don't think other wikis will use it but we can discuss it at the time we add the automation)
`autoOpponentImport` (to enable/disable the auto opponent import) <-- this one should probably be a standard option once we add the automation^^

Reply:
Config has been reworked to allow new to be added via `addCustomConfig` prior to running the input parsing.

Reply:
we can come back to this when automation is added in some weeks and sc2 actually would try to use the standardized prize pool